include/deal.II/base/quadrature.h

```diff
@@ -221,6 +221,19 @@ class Quadrature : public Subscriptor
   template <class Archive>
   void serialize (Archive &ar, const unsigned int version);
 
+  /**
+   * This function returns true if the quadrature object is a tensor product
+   * of one-dimensional formulas.
+   */
+  bool is_tensor_product() const;
+
+  /**
+   * In case the quadrature formula is a tensor product, this function
+   * returns the one-dimensional basis objects.
+   * Otherwise, calling this function is not allowed.
+   */
+  const std::vector<Quadrature<1>> &get_tensor_basis() const;
```

Comment:
can you make this a `std::array<Quadrature<1>, dim>`? Same for the member variable below.

Point being, we know the size of the array.